pydra/engine/core.py

```diff
@@ -504,15 +504,24 @@ def get_input_el(self, ind):
             return None, inputs_dict
 
     def to_job(self, ind):
-        """Run interface one element generated from node_state."""
+        """ Pickling tasks and inputs, so don't have to load the input to teh memory """
         # logger.debug("Run interface el, name={}, ind={}".format(self.name, ind))
-        el = deepcopy(self)
-        el.state = None
-        # dj might be needed
-        # el._checksum = None
-        _, inputs_dict = self.get_input_el(ind)
-        el.inputs = attr.evolve(el.inputs, **inputs_dict)
-        return el
+        # copy the task and setting input/state to None
+
+        pkl_files = self.cache_dir / "pkl_files"
+        pkl_files.mkdir(exist_ok=True)
+        task_main_path = pkl_files / f"task_main_{self.name}.pklz"
```

Comment:
how will this deal with name clashes across workflow hierarchy? one of the reasons for suggesting doing this at the individual split level instead of the whole job is so that you can reuse the file if it already exists. 

should also use filelock to write the file to avoid clashes.